src/nci_state_listen_active.c

```diff
@@ -0,0 +1,215 @@
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "nci_sm.h"
+#include "nci_state_impl.h"
+#include "nci_log.h"
+
+typedef NciState NciStateListenActive;
+typedef NciStateClass NciStateListenActiveClass;
+
+GType nci_state_listen_active_get_type(void) NCI_INTERNAL;
+#define THIS_TYPE (nci_state_listen_active_get_type())
+#define PARENT_CLASS (nci_state_listen_active_parent_class)
+
+G_DEFINE_TYPE(NciStateListenActive, nci_state_listen_active, NCI_TYPE_STATE)
+
+/*==========================================================================*
+ * Implementation
+ *==========================================================================*/
+
+static
+gboolean
+nci_state_listen_active_interface_error_ntf(
+    NciState* self,
+    const GUtilData* payload)
+{
+    const guint8* pkt = payload->bytes;
+    const guint len = payload->size;
+
+    /*
+     * Table 19: Control Messages for Interface Error
+     *
+     * CORE_INTERFACE_ERROR_NTF
+     *
+     * +=========================================================+
+     * | Offset | Size | Description                             |
+     * +=========================================================+
+     * | 0      | 1    | Status                                  |
+     * | 1      | 1    | Conn ID                                 |
+     * +=========================================================+
+     *
+     * 5.2.6 State RFST_LISTEN_ACTIVE
+     *
+     * ...
+     * When using the ISO-DEP or NFC-DEP RF interface, and the NFCC
+     * detects an error during the RF communication, which does not
+     * require returning to the IDLE state, as defined in the Listen
+     * Mode state machine, the NFCC SHALL send CORE_INTERFACE_ERROR_NTF,
+     * using the appropriate status out of RF_TRANSMISSION_ERROR,
+     * RF_PROTOCOL_ERROR and RF_TIMEOUT_ERROR. The state will then
+     * remain RFST_LISTEN_ACTIVE.
+     */
+    if (len == 2) {
+        NciSm* sm = nci_state_sm(self);
+
+        switch (pkt[0]) {
+        case NCI_RF_TRANSMISSION_ERROR:
+            GDEBUG("CORE_INTERFACE_ERROR_NTF (Transmission Error)");
+            nci_sm_switch_to(sm, NCI_RFST_DISCOVERY);
+            return TRUE;
+        case NCI_RF_PROTOCOL_ERROR:
+            GDEBUG("CORE_INTERFACE_ERROR_NTF (Protocol Error)");
+            nci_sm_switch_to(sm, NCI_RFST_DISCOVERY);
+            return TRUE;
+        case NCI_RF_TIMEOUT_ERROR:
+            GDEBUG("CORE_INTERFACE_ERROR_NTF (Timeout)");
+            nci_sm_switch_to(sm, NCI_RFST_DISCOVERY);
+            return TRUE;
+        }
+    }
+    /* Unrecornized notification */
```

Comment:
Typo, Unrecornized -> Unrecognized.

Reply:
There are more of those :) Will grep and fix.

Reply:
Fixed this one